File: 실습/4장/02_nested_inner_class/Main.java

public class Main {

    static class Outer {
        private final String text;

        public Outer(String text) {
            this.text = text;
        }

        static class Nested {
            public void printMessage() {
                System.out.println("Nested 클래스");
            }
        }

        class Inner {
            public void printOuterText() {
                System.out.println(text);
            }
        }
    }

    public static void main(String[] args) {
        Outer.Nested nested = new Outer.Nested();
        nested.printMessage();

        Outer outer = new Outer("Outer 클래스");
        Outer.Inner inner = outer.new Inner();
        inner.printOuterText();
    }
}

File: 실습/4장/02_nested_inner_class/Main.kt
class Outer(
    private val text: String
) {
    class Nested {
        fun printMessage() {
            println("Nested 클래스")
        }
    }

    inner class Inner {
        fun printOuterText() {
            println(text)
        }
    }
}

fun main() {
    val nested = Outer.Nested()
    nested.printMessage()

    val outer = Outer("Outer 클래스")
    val inner = outer.Inner()
    inner.printOuterText()
}